File: capgemini/batch-2/twoPointer.py



arr = [2,2,-9,3,6,6,3,7,-98,34,54]
target = 232 



def isPairExist(arr,target):
    arr.sort() # nlong
    i,j = 0, len(arr)-1
    while i<j: # n 
        if arr[i]+arr[j] > target:j=j-1
        if arr[i]+arr[j] < target:i=i+1
        if arr[i]+arr[j] ==target: return True
    return False


def isTripletExists(arr,target):
    for i in range(len(arr)):
        selectedEle = arr[i]
        subArr = arr[0:i]+arr[i+1:]
        updatedTarget = target-selectedEle
        ans = isPairExist(subArr,updatedTarget)
        if ans == True:
            return True
    return False



print(isTripletExists(arr,target))
#print(isPairExist(arr,target))
    

File: madurai/S27/26-06-2024/TwoPointer.java
import java.util.Arrays;

public class TwoPointer {
    public static void main(String[] args) {
        int[] arr = {-3,1,-5,3,-6,12,24,45,76,87,-87};
        int target = 1000;
        boolean ans = isPairExist(arr,target);
        System.out.println(ans);
    }

    private static boolean isPairExist(int[] arr, int target) {
        Arrays.sort(arr);
        int i=0, j= arr.length-1;


        while (i<j) {
            if(arr[i]+arr[j] == target) return true;
            if(arr[i]+arr[j] < target) i+=1 ;
            if(arr[i]+arr[j] > target) j-=1 ;
        }
        return false;

    }
}
